deployments/grafana-demo/02-auto-instrument.yml

```diff
@@ -0,0 +1,162 @@
+        traces  = [otelcol.exporter.otlp.tempo.input]
+      }
+    }
+      
+    otelcol.exporter.prometheus "default" {
+      forward_to = [prometheus.remote_write.mimir.receiver]
+    }
+      
+    prometheus.remote_write "mimir" {
+      endpoint {
+        url = "https://" + env("MIMIR_USER") + ":" + env("GRAFANA_API_KEY") + "@" + env("MIMIR_ENDPOINT") + "/api/prom/push"
+      }
+    }
+      
+    otelcol.exporter.otlp "tempo" {
+      // Send traces to a locally running Tempo without TLS enabled.
+      client {
+        endpoint = env("TEMPO_ENDPOINT")
+        auth     = otelcol.auth.basic.creds.handler
+      }
+    }
+      
+    otelcol.auth.basic "creds" {
+      username = env("TEMPO_USER")
+      password = env("GRAFANA_API_KEY")
+    }
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: grafana-agent
+  labels:
+    app: grafana-agent
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: grafana-agent
+  template:
+    metadata:
+      labels:
+        app: grafana-agent
+    spec:
+      volumes:
+        - name: grafana-agent-config
+          configMap:
+            name: grafana-agent-config
+      containers:
+        - name: grafana-agent
+          image: grafana/agent:main
+          command:
+            - "/usr/bin/agent"
+            - "run"
+            - "/grafana-agent-config/agent-config.river"
+          env:
+            - name: AGENT_MODE
+              value: flow
+            - name: TEMPO_ENDPOINT
+              valueFrom:
+                secretKeyRef:
+                  key: tempo-endpoint
+                  name: grafana-secret
+            - name: TEMPO_USER
+              valueFrom:
+                secretKeyRef:
+                  key: tempo-user
+                  name: grafana-secret
+            - name: MIMIR_ENDPOINT
+              valueFrom:
+                secretKeyRef:
+                  key: mimir-endpoint
+                  name: grafana-secret
+            - name: MIMIR_USER
+              valueFrom:
+                secretKeyRef:
+                  key: mimir-user
+                  name: grafana-secret
+            - name: GRAFANA_API_KEY
+              valueFrom:
+                secretKeyRef:
+                  key: grafana-api-key
+                  name: grafana-secret
+          ports:
+            - containerPort: 4318
+              protocol: TCP
+              name: http-traces
+            - containerPort: 4317
+              protocol: TCP
+              name: grpc-traces
+          volumeMounts:
+            - mountPath: /grafana-agent-config
+              readOnly: true
+              name: grafana-agent-config
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: grafana-agent
+spec:
+  selector:
+    app: grafana-agent
+  ports:
+    - name: http
+      port: 4318
+      protocol: TCP
+      targetPort: http-traces
+    - name: grpc
+      port: 4317
+      protocol: TCP
+      targetPort: grpc-traces
+---
+apiVersion: apps/v1
+kind: DaemonSet
```

Comment:
Why not deploy it as a sidecar container in the goblog deployment? 
Deploying it as a DaemonSet would only instrument a single process even if you deploy multiple instances of the goblog application.